FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +20,50 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard let _ = self else { return }
+			switch result {
+			case let .success((data, response)):
+				guard response.statusCode == RemoteFeedLoader.successStatusCode else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+				guard let feed = try? JSONDecoder().decode(Feed.self, from: data) else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+				completion(.success(feed.feedImages))
+			case .failure:
+				completion(.failure(Error.connectivity))
+			}
+		}
+	}
+
+	private struct Feed: Decodable {
+		let items: [FeedImageAPI]
+
+		var feedImages: [FeedImage] {
+			items.compactMap { $0.feedImage }
```

Comment:
If `$0.feedImage` returns `nil`, we are supposed to complete with `.invalidData` instead of ignoring that particular item using `.compactMap`. In our case that might happen because of an invalid `url`, so we should deal with invalid urls during mapping.

Please replace `compactMap` with `map`:
```suggestion
			items.map { $0.feedImage }
```